src/tracers.jl

```diff
@@ -1,30 +1,19 @@
 abstract type AbstractTracer <: Real end
 
-sparse_vector(T, index::Integer) = T([index])
-
 #===================#
 # Set operations    #
 #===================#
 
-myempty(::Type{S}) where {S<:AbstractSet} = S()
-
-function clever_union(a::AbstractSet{I}, b::AbstractSet{I}) where {I}
-    if isempty(a)
-        return b
-    elseif isempty(b)
-        return a
-    else
-        return union(a, b)
-    end
-end
+@inline myempty(::Type{S}) where {S<:AbstractSet} = S()
+@inline seed(::Type{S}, i::Integer) where {S<:AbstractSet} = S([i])
```

Comment:
I wonder if we can be smarter than by creating a vector and then a set

Reply:
We have `isempty` flags now, so we can finally use `undef`. I will do this in a follow-up PR today, I didn't want to make benchmarks uninformative by touching too many things that modify the performance.